code/__HELPERS/_string_lists.dm

```diff
@@ -1,27 +1,46 @@
 #define pick_list(FILE, KEY) (pick(strings(FILE, KEY)))
+#define pick_list_replacements(FILE, KEY) (strings_replacement(FILE, KEY)))
 
 var/global/list/string_cache
 
+/proc/strings_replacement(filename, key)
+	load_strings_file(filename)
+
+	if((filename in string_cache) && (key in string_cache[filename]))
+		var/response = pick(string_cache[filename][key])
+		//Loop through the line, finding any replacement elements needing replacements
+		//defined by @pick(key), the sub key is assumed to be in the same file
+		//and will use pick_list to find and pick it's random value. At this time
+		//further subkey replacements are not yet supported
+		var/start = findtext('@pick(')
+		var/end = 0
+		var/sub_key = ''
+		var/lookup_val = ''
+		while(start)
+			end = findtext(')', start)
+			sub_key = copytext(response, start+6, end)
+			lookup_val = pick_list(filename, sub_key)
+			world << lookup_val
+			start = findtext('@pick(')

```

Comment:
Are people of the opinion that regex is faster than straight findtext hacks like this? Have any performance tests been done?